src/js/disability-benefits/526EZ/components/ReviewCardField.jsx

```diff
@@ -0,0 +1,212 @@
+
+import { errorSchemaIsValid } from '../../../common/schemaform/validation';
+
+import set from '../../../../platform/utilities/data/set';
+import get from '../../../../platform/utilities/data/get';
+
+
+/**
+ * Displays a review card if the information inside is valid.
+ *
+ * For use on a schema of type 'object' or 'array'.
+ * Intended to wrap objects or arrays to avoid duplicate functionality here.
+ */
+export default class ReviewCardField extends React.Component {
+  static defaultProps = {
+    uiSchema: {},
+    errorSchema: {},
+    idSchema: {},
+    registry: getDefaultRegistry(),
+    required: false,
+    disabled: false,
+    readonly: false,
+  }
+
+
+  constructor(props) {
+    super(props);
+
+    // Throw an error if there’s no viewComponent (should be React component)
+    if (typeof get('ui:options.viewComponent', this.props.uiSchema) !== 'function') {
+      throw new Error(`No viewComponent found in uiSchema for ReviewCardField ${this.props.idSchema.$id}.`);
+    }
+
+    const acceptedTypes = ['object', 'array'];
+    if (!acceptedTypes.includes(this.props.schema.type)) {
+      throw new Error(`Unknown schema type in ReviewCardField. Expected one of [${acceptedTypes.join(', ')}], but got ${this.props.schema.type}.`);
+    }
+
+    this.state = {
+      // Set initial state based on whether all the data is valid
+      editing: !errorSchemaIsValid(props.errorSchema)
+    };
+  }
+
+  onPropertyChange(name) {
+    return (value) => {
+      const formData = Object.keys(this.props.formData || {}).length
+        ? this.props.formData
+        : getDefaultFormState(this.props.schema, undefined, this.props.registry.definitions);
+      this.props.onChange(set(name, value, formData));
+    };
+  }
+
+
+  getTitle = () => {
+    const { uiSchema, formData } = this.props;
+    return typeof uiSchema['ui:title'] === 'function'
+      ? uiSchema['ui:title'](formData)
+      : uiSchema['ui:title'];
+  }
+
+
+  /**
+   * Much of this is taken from ArrayField & ObjectField.
+   *
+   * Renders a SchemaField for each property it wraps.
+   */
+  getEditView = () => {
+    const {
+      disabled,
+      errorSchema,
+      formData,
+      idSchema,
+      onBlur,
+      readonly,
+      registry,
+      schema,
+      uiSchema
+    } = this.props;
+    const { SchemaField } = registry.fields;
+
+    let properties;
+    if (schema.type === 'object') {
+      properties = Object.keys(schema.properties);
+    } else if (schema.type === 'array') {
+      properties = Object.keys(schema.items.properties);
+    } else {
+      // Shouldn't get here, but in case we update the gate in the constructor later...
+      throw new Error(`Unknown schema type in ReviewCardField: ${schema.type}`);
+    }
+
+    const title = this.getTitle();
+
+    return (
+      <div className="review-card">
+        <div className="review-card--body input-section va-growable-background">
+          <h4>{title}</h4>
+          {properties.map(propName => {
```

Comment:
I don't think you need this mapping. I think you can call SchemaField with the props passed to this custom field. That would also let you get rid of `onPropertyChange`. By choosing to call SchemaField for each property in your object, you're forcing yourself into duplicating the `ObjectField` component.